doc/usage.rst

```diff
@@ -109,6 +114,37 @@ On initialization, the PennyLane-Forest devices accept additional keyword argume
     on a provided QMI, these environment variables are set automatically and will also
     not need to be passed in PennyLane.
 
+.. note::
+
+    If using the pyQuil built-in pyQVM, you will still need to have an accessible Forest Quil compiler
+    server. This must be launched with the command ``quilc -S -P``, where the ``-P`` flag restricts
+    the input/output of the compiler to protoquil.
```

Comment:
Oh, yeah that paragraph is not entirely clear. From what I can tell, the NumpyWavefunctionSimulator works completely independently of Quilc. However, `ReferenceWavefunctionSimulator` and `ReferenceDensitySimulator` both seem to require Quilc (my program hangs if Quilc in protoquil mode is not available).

I couldn't find anything in the PyQuil documentation regarding this?

Reply:
Yes, unfortunately we don't really have good documentation yet for the `ReferenceWavefunctionSimulator` and `ReferenceDensitySimulator`, but they don't require Quilc to run. For example, I can run the following code without quilc (or the lisp qvm) running in the background:

```
from pyquil.reference_simulator import ReferenceWavefunctionSimulator, \
                                       ReferenceDensitySimulator
from pyquil import Program
from pyquil.gates import H, X


sim_wfn = ReferenceWavefunctionSimulator(n_qubits=2)
sim_rho = ReferenceDensitySimulator(n_qubits=2)

for instruction in [H(0), X(1), H(1)]:
    sim_wfn.do_gate(instruction)
    sim_rho.do_gate(instruction)

print(f"Wavefunction: {sim_wfn.wf}")
print(f"Density Matrix: {sim_rho.density}")
```
which outputs
```
Wavefunction: [ 0.5+0.j  0.5+0.j -0.5+0.j -0.5+0.j]
Density Matrix: [[ 0.25+0.j  0.25+0.j -0.25+0.j -0.25+0.j]
 [ 0.25+0.j  0.25+0.j -0.25+0.j -0.25+0.j]
 [-0.25+0.j -0.25+0.j  0.25+0.j  0.25+0.j]
 [-0.25+0.j -0.25+0.j  0.25+0.j  0.25+0.j]]
```

Reply:
I have to admit, this still confuses me a bit, mostly from this issue here: https://github.com/rigetti/pyquil/issues/803

Since the pyQVM should be called primarily through `get_qc`, which simulator (`NumpyWavefunctionSimulator`, `ReferenceWavefunctionSimulator`, or `ReferenceDensitySimulator`) would be used in the following example?

```python
get_qc('2q-pyqvm')
```

From my tests, and from the issue linked above, it appears that quilc running in protoquil mode is required, otherwise the command hangs. QVM is definitely not needed, though.

Reply:
In the case of `get_qc('2q-pyqvm')`, it will return a `NumpyWavefunctionSimulator`. If you specify post gate noise probabilities, it will return a `ReferenceDensitySimulator` (see [here](https://github.com/rigetti/pyquil/blob/master/pyquil/pyqvm.py#L224)). Protoquil just refers to quil without classical controls. The pyQVM doesn't actually need the quil compiler to parse those quil instructions, which is handled by numpy under the hood.

Reply:
Ah, I see what is happening now. In the QVM method, the `prog.compile()` method is being explicitly called, which then causes the PyQVM simulator to attempt to compile via `quilc`. This has been modified in the `qvm.py` file; if the quantum device is the `pyqvm`, then `qc.run(prog)` is called directly, skipping compilation.

As of the latest commit, everything now works and passes (tested against the latest qvm, quilc, and pyquil).